### PR TITLE
Share icons enhancements

### PR DESCRIPTION
**Pull request by arinaturcu**
Adds the following settings to the Sharing Icons block:
- Items gap
- Border radius
- Toggle to disable underlined text
- Color settings for background and text

### CODE REVIEW

**Comment by arinaturcu** [comment]
@HardeepAsrani The issue here is that when updating from the current version to this one, the CSS is not loaded until the Update button is pressed in the Editor.
Can you give me some insights on how the CSS files are loaded and how to solve this?

**Comment by irinelenache** [comment]
@arinaturcu Good job 🚀 , tested and here are my findings:

- [ ] After previously adding the block and updating to this version, the block lost all spacings (I think it has something to do with the alignments, because removing the **align center** option will restore the spacings)
before update: https://vertis.d.pr/2GTMmx
after update: https://vertis.d.pr/LXHKIB

- [ ] When the icons are center aligned, the separator between the icon and text is lost https://vertis.d.pr/494hlm
- [ ] If i raise the Spacing gap of the icons with **regular style**, the text will get out of the container https://vertis.d.pr/SsQNn3
- [ ] The blocks don't look ok on mobile, even with default settings
desktop: https://vertis.d.pr/86oZBY
mobile: https://vertis.d.pr/DDGgZE

- [ ] Changing the block alignment will affect the other blocks positions, both in editor and frontend. Here is a video with the issue https://vertis.d.pr/kJifrP

The issues can be checked on this instance:
```
 Admin area URL: https://wellmadeaftermath.s4-tastewp.com/wp-admin 
 Username: irinel 
 Password: YPyXoYEVMC8 
```

File: src/blocks/blocks/sharing-icons/inspector.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		let newAttributes = {};
+		if ( value === undefined ) {
+			newAttributes[name] = metadata.attributes[name].default;
+			setAttributes( newAttributes );
+			return;
+		}
+
+		newAttributes[name] = `${ value }px`;
+		setAttributes( newAttributes );
+	};
```
[the remark this answers is not in the document]
Comment (arinaturcu) [reply src/blocks/blocks/sharing-icons/inspector.js:35]: You're right, I didn't know I can use `'unit'` :sweat_smile:

**Comment by arinaturcu** [comment]
@irinelenache Thank you for testing! :rocket: 
Things should work fine now.


> * [x]  Changing the block alignment will affect the other blocks positions, both in editor and frontend. Here is a video with the issue https://vertis.d.pr/kJifrP

This one in fixed in the frontend. It seems that this was happening in the old version too, so I am not sure if it supposed to work like this or not. Can you take a look, @HardeepAsrani?

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
@arinaturcu This will need some more work.

Few things I noticed:

- We don't need an option to remove text underline. We can just add CSS to not have it by default.
- For Gap & Radius, once again the defaults can be in the CSS, not the attributes.
- The option to customize color/text is needed for each button separately, not one for all. We can show them under one panel and only when the button is visible.

**Comment by HardeepAsrani** [comment]
Any updates on this @arinaturcu?

**Comment by arinaturcu** [comment]
@HardeepAsrani I am working on the color control for the individual icons

**Comment by arinaturcu** [comment]
@HardeepAsrani Everything is ready here. Can you review this when you get some time?

**Review by HardeepAsrani — APPROVED** [review]

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
If you migrate from an older version, for example if you paste this in the HTML editor:

`<!-- wp:themeisle-blocks/sharing-icons {"reddit":true,"className":"is-style-icons"} /-->`

You can this debugging error:

`PHP Notice:  Undefined index: id in /var/www/html/wp-content/plugins/otter-blocks/inc/render/class-sharing-icons-block.php on line 91`

So you need to make sure the new attribute structure works well in the frontend too.

**Comment by HardeepAsrani** [comment]
@JohnPixle Can you take a look and see if all your requested changes are properly addressed in this PR? You can download the plugin from the first comment. Thank you!

**Review by HardeepAsrani — APPROVED** [review]

**Comment by JohnPixle** [comment]
@HardeepAsrani Sorry, I tried to download the file from the first comment, and tested it on local, but obviously something is broken, tried to regenerate the CSS, but nothing. It's definitely not showing up as in @irinelenache 's video.

I downloaded the dev version of otter blocks.

Frame 514

I see you've been working on it, perhaps I got in the middle of a file update.

Thanks!

**Comment by HardeepAsrani** [comment]
@JohnPixle I forgot and merged the PR. You can find the file from here: https://verti-artifacts.s3.amazonaws.com/otter-blocks-development-ffda03f3/otter-blocks.zip - let me know if it works properly.

**Comment by JohnPixle** [comment]
Thanks, worked now @HardeepAsrani 👍🏻 

I'd say it's a good update for the block, I can see a couple of more improvements regarding styles, probably in one in the next versions. Will follow up accordingly.

Thanks, looking good.!

**Comment by JohnPixle** [comment]
Oh, sorry, I forgot to mention that the slider for border radius might need some "calibration". From value 50 until 100, (half slider) there is no visible change in the style. Value 100 should be the totally rounded radius, now it gets rounded in 50.
Screenshot 2022-05-23 at 19 34 43
.

**Comment by arinaturcu** [comment]
@JohnPixle Got it. Also, I just noticed a big typo in 'Border Badius' :sweat_smile:

**Comment by JohnPixle** [comment]
@arinaturcu OMG!  I'd say it's a valid observation! 👍🏻

**Comment by HardeepAsrani** [comment]
I've merged them so thank you both. I think it'll be a good exercise to get the PRs reviewed by the design team before merging them especially if it's a big rehaul of UI.

Thanks, both!

**Comment by JohnPixle** [comment]
Awesome, yes I agree. Please also make sure to either ping @Codeinwp/design-team or any of us individually when you create a ticket that involves design.

Thanks all!

**Pull request merged**